packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -171,20 +184,20 @@ export class AttributionCollection implements IAttributionCollection<Attribution
     ): SerializedAttributionCollection {
         const posBreakpoints: number[] = [];
         const seqs: number[] = [];
-        let mostRecentAttributionKey: number | undefined;
+        let mostRecentAttributionKey: AttributionKey | undefined;
         let cumulativePos = 0;
 
         let segmentsWithAttribution = 0;
         let segmentsWithoutAttribution = 0;
         for (const segment of segments) {
             if (segment.attribution) {
                 segmentsWithAttribution++;
-                for (const { offset, key: info } of segment.attribution?.getAll() ?? []) {
-                    if (info.seq !== mostRecentAttributionKey) {
+                for (const { offset, key } of segment.attribution?.getAll() ?? []) {
+                    if (!mostRecentAttributionKey || !areEqualAttributionKeys(key, mostRecentAttributionKey)) {
                         posBreakpoints.push(offset + cumulativePos);
-                        seqs.push(info.seq);
+                        seqs.push(key.seq);
                     }
-                    mostRecentAttributionKey = info.seq;
```

Comment:
LGTM. One minor concern about the variable naming is that in previous implementation `info` means the `AttributionKey: { type: "op", seq: xxx }`. However, in the latest update `info` represents the sequence number instead, the `AttributionKey` is `{ type: "op", seq: **currentInfo** }`. It made me hesitate at the first glance but it is nothing serious

Reply:
yeah, i adjusted some of the naming here a bit. i think it's clearer now, but you're right it's not a purely mechanical change. i'll probably switch things more in future PRs, so going to leave this as-is for now.